### ISSUE TITLE
CA1822 - Support for TUnit

### ISSUE THREAD

**Issue by viceroypenguin** [dotnet/roslyn-analyzers#7453]
### Analyzer

**Diagnostic ID**: [CA1822](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1822)

### Describe the improvement

Currently, CA1822 provides exclusions when attributes from XUnit, MSTest, and NUnit are applied to a method. I am in the process of migrating to TUnit, a new testing framework; but since it does not use the same attributes, `[Test]` methods are being flagged by CA1822.

It would be helpful to either/both:
1. have a parameter in the .editorconfig to add additional attributes to the skipped list
2. add `TUnit.Core.TestAttribute`, `TUnit.Core.BeforeAttribute`, `TUnit.Core.BeforeEveryAttribute, `TUnit.Core.AfterAttribute`, `TUnit.Core.AfterEveryAttribute` to the exclusion list.

### Describe suggestions on how to achieve the rule

1. Am not familiar enough with the process to provide clear directions on how to improve. Expected .editorconfig might look something like:
```
dotnet_code_quality.CA1822.excluded_attributes = T:TUnit.Core.TestAttribute, T:TUnit.Core.BeforeAttribute
```

2. 
   * The above listed attributes added to `WellKnownTypeNames`
   * The above listed attributes added at: https://github.com/dotnet/roslyn-analyzers/blob/main/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.cs

### Additional context

TUnit documentation: https://thomhurst.github.io/TUnit/docs/intro

**Comment by FelixDamrau** [comment]
Is there any chance this issue can get fixed?
The linked PR of the TUnit author looked fine to me but was closed without any explanation (that I can see).

Is this something that should just not get merged?
Can I do something to make a fix more likely? I'd be happy to help.
